src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowMarkdownCommand.cs

```diff
@@ -0,0 +1,166 @@
+    /// Html will be displayed in default browser.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Show, "Markdown",
+        HelpUri = "TBD")]
+    [OutputType(typeof(string))]
+    public class ShowMarkdownCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Gets or sets InputObject of type Microsoft.PowerShell.MarkdownRender.MarkdownInfo to display.
+        /// </summary>
+        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch to view Html in default browser.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter UseBrowser { get; set; }
+
+        private SteppablePipeline stepPipe;
+
+        /// <summary>
+        /// Override BeginProcessing.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (!this.MyInvocation.BoundParameters.ContainsKey("UseBrowser"))
+            {
+                // Since UseBrowser is not bound, we use proxy to Out-Default
+                stepPipe = ScriptBlock.Create(@"Microsoft.PowerShell.Core\Out-Default @PSBoundParameters").GetSteppablePipeline(this.MyInvocation.CommandOrigin);
+                stepPipe.Begin(this);
+            }
+        }
+
+        /// <summary>
+        /// Override ProcessRecord.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            object inpObj = InputObject.BaseObject;
+            var markdownInfo = inpObj as MarkdownInfo;
+
+            if (markdownInfo == null)
+            {
+                string errorMessage = StringUtil.Format(ConvertMarkdownStrings.InvalidInputObjectType, inpObj.GetType());
+                var errorRecord = new ErrorRecord(
+                            new ArgumentException(errorMessage),
+                            "InvalidInputObject",
+                            ErrorCategory.InvalidArgument,
+                            InputObject);
+
+                WriteError(errorRecord);
+            }
+            else
+            {
+                if (UseBrowser)
+                {
+                    var html = markdownInfo.Html;
+
+                    if (!string.IsNullOrEmpty(html))
+                    {
+                        string tmpFilePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + ".html");
+                        using (var writer = new StreamWriter(new FileStream(tmpFilePath, FileMode.Create, FileAccess.Write, FileShare.Write)))
+                        {
+                            writer.Write(html);
+                        }
+
+                        if (outputBypassTestHook)
+                        {
+                            WriteObject(html);
+                            return;
+                        }
+
+                        ProcessStartInfo startInfo = new ProcessStartInfo();
```

Comment:
Does something reasonable happen on non-Windows systems?

Reply:
I tested it, UseShellExecute works on non-Windows now.